### ISSUE TITLE
hdevtools is looking for the stack executable in a non-stack project

### ISSUE THREAD

**Issue by expipiplus1** [hdevtools/hdevtools#35]
`hdevtools --start-server` prints `/bin/sh: 1: stack: not found`

To reproduce:

```
mkdir foo
cd foo
cabal init
<select the defaults on all the options, it doesn't matter what one selects on Library/Executable>
hdevtools --start-server
```

There is no `stack.yaml` present yet hdevtools still tries to call `stack`. If I put an empty executable file named `stack` somewhere in `$PATH` then hdevtools doesn't work at all.

**Comment by expipiplus1** [comment]
The fix seems to be checking for the existence of the `stack` executable in `findStackYaml` before trying to call it.

**Comment by ch1bo** [comment]
Good catch, thanks. I might be able to fix this today.

**Comment by expipiplus1** [comment]
Thanks!

**Comment by ch1bo** [comment]
Merged with 6bc57ebbf5c71f446e2ec11536d2b8ad4a6ba75d
